### ISSUE TITLE
VB version of TestDuplicateConstantAttributesMetadata is locale dependent

### ISSUE THREAD

**Issue by Paxxi** [dotnet/roslyn#28830]
**Version Used**: Master

**Steps to Reproduce**:

1. Set Windows to use Swedish locale for date/time and number settings
2. restore.cmd
3. build.cmd
4. test.cmd
Alternatively
2. restore.cmd
3. build in VS
4. Run the VB test `TestDuplicateConstantAttributesMetadata` only

**Expected Behavior**:
Test should pass like the C# version does.
**Actual Behavior**:
 Test fails with the following
```
Test Name:	Microsoft.CodeAnalysis.VisualBasic.UnitTests.Emit.OptionalArgumentsTests.TestDuplicateConstantAttributesMetadata
Test FullName:	Microsoft.CodeAnalysis.VisualBasic.UnitTests.Emit.OptionalArgumentsTests.TestDuplicateConstantAttributesMetadata
Test Source:	C:\code\roslyn\src\Compilers\VisualBasic\Test\Emit\Emit\OptionalArgumentsTests.vb : line 735
Test Outcome:	Failed
Test Duration:	0:00:03,188

Result StackTrace:	
at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Execute(String moduleName, String[] args, String expectedOutput) in C:\code\roslyn\src\Test\Utilities\Portable\Platform\Desktop\DesktopRuntimeEnvironment.cs:line 245
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in C:\code\roslyn\src\Test\Utilities\Portable\CompilationVerifier.cs:line 114
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in C:\code\roslyn\src\Test\Utilities\Portable\CommonTestBase.cs:line 154
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in C:\code\roslyn\src\Test\Utilities\Portable\CommonTestBase.cs:line 68
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in C:\code\roslyn\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 73
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(Compilation compilation, XCData expectedOutput, String[] args, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, EmitOptions emitOptions, Verification verify) in C:\code\roslyn\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 101
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.Emit.OptionalArgumentsTests.TestDuplicateConstantAttributesMetadata() in C:\code\roslyn\src\Compilers\VisualBasic\Test\Emit\Emit\OptionalArgumentsTests.vb:line 826
Result Message:	
Roslyn.Test.Utilities.ExecutionException : 
Execution failed for assembly 'C:\Users\Perbj\AppData\Local\Temp\RoslynTests'.
Expected: 
System.Reflection.Missing: System.Reflection.Missing
System.DateTime: 01/01/0001 00:00:00
System.DateTime: 01/01/0001 00:00:00
System.DateTime: 01/01/0001 00:00:00
System.Int32: 0
System.Int64: 3
System.Decimal: 3

Actual:   System.Reflection.Missing: System.Reflection.Missing
System.DateTime: 01-01-0001 00:00:00
System.DateTime: 01-01-0001 00:00:00
System.DateTime: 01-01-0001 00:00:00
System.Int32: 0
System.Int64: 3
System.Decimal: 3
```
Notice the different separators used in the date format.

**Comment by Paxxi** [comment]
More info, testing this targeting .NET 4.6.1 gives the same incorrect results in both VB and C#
sample programs I tested with
```C#
using System;

namespace ConsoleApp2
{
    class Program
    {
        static void Main(string[] args)
        {
            var test = new DateTime(1, 1, 1, 0, 0, 0).ToString("MM/dd/yyyy HH:mm:ss");
            Console.WriteLine(test);
        }
    }
}
```
```VB
Module Module1
    Sub Main()
        Dim test = New DateTime(1, 1, 1, 0, 0, 0).ToString("MM/dd/yyyy HH:mm:ss")
        Console.WriteLine(test)
    End Sub
End Module
```
This appears to be a bug in .NET, when adding `/` randomly to the dateformat string they're all replaced with `-` even if it's not a known date format string.

**Comment by jaredpar** [comment]
> This appears to be a bug in .NET, when adding / randomly to the dateformat string they're all replaced with - even if it's not a known date format string.

Recently ran into this as well and it turns out it's by design. The `/` in the format string is not interpreted literally. It means instead to use the local specific date separator as `:` means the time separator. If you want to have a literal `/` then you need to use `'/'` in the format string.

**Comment by jaredpar** [comment]
Closing as a PR was merged that fixed this: #28849
